### PR TITLE
#8585 Add submitter information to provenance metadata

### LINKED ISSUE #8585: Add submitter information to provenance metadata
Items that are submitted to collections that allow immediate archiving are missing "Submitted by" text from provenance metadata.  This is closely related to https://github.com/DSpace/DSpace/issues/7667 

**Describe the solution you'd like**
Similar to items that go through a workflow, there should be provenance metadata that contains this text along with the -person who submitted the item.

**Describe alternatives or workarounds you've considered**
This was seemingly fixed in version 6.4, but not in version 7.x

**Additional context**
In XmlWorkflowServiceImpl.java, this is accomplished in the method recordStart().  However, this method is never reached when there is not a valid workflow step -- the item goes directly to the archive and the desired text is not added to provenance.

### PR DESCRIPTION
**Pull request by aroman-arvo**
## References
* Fixes #8585 

## Description
Added provenance with user information to submits without workflows

## Instructions for Reviewers
Make a submit to a collection without workflow and Provenance witt "Submitted by ..." information is present.

## Checklist

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [X] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [X] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [X] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [X] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [X] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [X] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by alanorth** [comment]
Thanks @aroman-arvo. Does this also cover the scenario where metadata is imported via `dspace metadata-import` from the command line? I notice in the past that items imported in bulk like this do not have provenance information.

**Comment by aroman-arvo** [comment]
@alanorth  Well seen. It does not take it into account. Nor the import of zips either.
It can be done easily but if you already have provenance in the data to be imported, an additional provenance should be created?

**Comment by alanorth** [comment]
> _It can be done easily but you already have provenance in the data to be imported, an additional provenance should be created?_

@aroman-arvo we import thousands of records via CSV import and we never include provenance information. It always seemed strange to me that the email address of the eperson is mandatory when doing a `dspace metadata-import`, but it is not recorded anywhere in the resulting records.

**Comment by aroman-arvo** [comment]
If you export from one DSpace and import into another, the data brings the original provenance data. Maybe we can simply add the new provenances, and traceability is preserved.
The email is necessary if the items are new, so they are assigned to that user as their owner.

**Comment by alanorth** [comment]
> _If you export from one DSpace and import into another, the data brings the original provenance data. Maybe we can simply add the new provenances, and traceability is preserved._

Not all imports come from other DSpaces. ;) For example, at our site we often digitize a bunch of archive records and create a CSV for import.

> _
> _The email is necessary if the items are new, so they are assigned to that user as their owner._

I thought the general principle was that DSpace should record provenance when it takes possession of an item. By default `dspace metadata-import` skips the workflow and then there is no provenance either.

**Comment by alanorth** [comment]
Thanks, @aroman-arvo! This works great for the command line import case. I tested on DSpace 7.6 by importing a simple item via `dspace metadata-import` before and after applying the patch.

Before, I only have one provenance field saying when the item was made available:

- dc.description.provenance | Made available in DSpace on 2023-10-17T07:25:07Z (GMT). No. of bitstreams: 0

After, I have two:

- dc.description.provenance | Submitted by Alan Orth (me@me.com) on 2023-10-17T07:29:05Z No. of bitstreams: 0
- dc.description.provenance | Made available in DSpace on 2023-10-17T07:29:05Z (GMT). No. of bitstreams: 0 | en

I am +1 by testing.

File: dspace-api/src/main/java/org/dspace/xmlworkflow/XmlWorkflowServiceImpl.java [hunk 1]
```diff
@@ -334,7 +336,7 @@ protected void activateFirstStep(Context context, Workflow wf, Step firstStep, X
                         + "item_id=" + wfi.getItem().getID()
                         + "collection_id=" + wfi.getCollection().getID()));
 
-        // record the start of the workflow w/provenance message
+// record the start of the workflow w/provenance message
```
Comment (alanorth) [inline dspace-api/src/main/java/org/dspace/xmlworkflow/XmlWorkflowServiceImpl.java]: Minor nitpick about this misaligned comment.

**Review by alanorth — APPROVED** [review]

**Pull request merged**